### DISCUSSION TITLE
[BUG] Having issues when trying to upload files to sharepoint online using add-PnPfile with a service principal made using azure portal

### DISCUSSION THREAD

**Discussion by skrrt16** [pnp/powershell#3298] [General]
### Reporting an Issue or Missing Feature
When trying to upload files using add-PnPfile with a service principal made using the azure portal, it fails and returns an error of unauthorized or access denied, if the service principal is created using register-PnPAzureADApp it works fine, NOTE: both service principal were given the same permissions.
### Expected behavior 
I expect it to upload files irregardless how the service principal is created.

### Actual behavior 
returns an unathorized or access denied when using service principal made in azure portal.

### Steps to reproduce behavior
create a service principal using azure active directory > app registration, and give it the necessary permissions
connect to sharepoint online using created service principal.
upload files using add-pnpfile

### What is the version of the Cmdlet module you are running?
ModuleType Version    Name                                ExportedCommands                                         
---------- -------    ----                                ----------------                                         
Manifest   1.12.0     PnP.PowerShell                      {Add-PnPAdaptiveScopeProperty, Add-PnPPropertyBagValue...

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
Must be some missing configuration or something else ? 
If it is unauthorized, it means there's something else that is missing when you create from Azure portal.
Check the difference between the one created by PnP and one by Azure portal, maybe something is different.
Moving to discussion.
